test/jdk/com/sun/jdi/TestScaffold.java

```diff
@@ -549,11 +547,12 @@ private ArgInfo parseArgs(String args[]) {
             }
         }
 
-        // Need to change args to run wrapper using command like 'TestScaffold Virtual <app-name>'
-        String mainWrapper = System.getProperty("main.wrapper");
+        // Need to change args to run wrapper using command like 'DebuggeeWrapper Virtual <app-name>'
+        // Additionally property 'main.wrapper' should be set for tests which start additional threads
+        String mainWrapper = DebuggeeWrapper.getWrapperName();
         if (mainWrapper != null && !argInfo.targetAppCommandLine.isEmpty()) {
-            argInfo.targetVMArgs += "-Dmain.wrapper=" + mainWrapper;
-            argInfo.targetAppCommandLine = TestScaffold.class.getName() + ' '
+            argInfo.targetVMArgs += "-D" + DebuggeeWrapper.PROPERTY_NAME + "=" + mainWrapper;
+            argInfo.targetAppCommandLine = DebuggeeWrapper.class.getName() + ' '
```

Comment:
I know this is a pre-existing issue, but it just seems strange that we have to both set the main.wrapper property to "Virtual" and also pass "Virtual" as the first argument to DebuggeeWrapper.main(). Could we leave it up to DebuggeeWrapper.main() to set main.wrapper when the first arg is "Virtual"?

Reply:
It may break the tests that check this property in static initializers of debugee. In such case method could be called before main() and return an empty value if a property is not set yet.

Probably the DebuggerWrapper requires more documentation about it's usage. I could add it separately with refactoring.,

Reply:
Is it necessary to pass the wrapper argument if we are setting main.wrapper property? Can't DebuggeeWrapper just pick up the wrapper type from the property instead of the wrapper argument?

Reply:
That's reasonable.